pkg/cache/clusterqueue.go

```diff
@@ -225,14 +225,24 @@ func (c *clusterQueue) update(in *kueue.ClusterQueue, resourceFlavors map[kueue.
 
 func filterFlavorQuantities(orig resources.FlavorResourceQuantities, resourceGroups []kueue.ResourceGroup) resources.FlavorResourceQuantities {
 	ret := make(resources.FlavorResourceQuantities, len(orig))
+
+	for key, value := range orig {
+		if value > 0 {
+			ret[key] = value
+		}
+	}
+
 	for _, rg := range resourceGroups {
```

Comment:
We shouldn't need to differentiate between zero value and absent value for `usage`. Can we delete this logic, and rename the function to something like "filterZeros"? The only annoying part is this requires test updates...

Furthermore, I'm tempted to delete this function altogether [1] - and be okay leaking memory for a few int64s

[1] and also update newClusterQueue to allocate these maps, as we rely on this function for allocating the maps

Reply:
> Furthermore, I'm tempted to delete this function altogether [1] - and be okay leaking memory for a few int64s. 

I thought about it too. Just don't know is it using keys somewhere. Will check.

Reply:
I removed `filterLocalQueueUsage` and fixed tests. Looks like working find. Thanks.

Reply:
Looking great without the `filterLocalQueueUsage` approach.
Thanks.